staging/src/k8s.io/apiserver/pkg/admission/config.go

```diff
@@ -101,10 +100,10 @@ func ReadAdmissionConfiguration(pluginNames []string, configFilePath string, con
 	if err2 := yaml.Unmarshal(data, &unstructuredData); err2 != nil {
 		return nil, err
 	}
-	_, isLegacyImagePolicy := unstructuredData["imagePolicy"]
-	_, isLegacyPodNodeSelector := unstructuredData["podNodeSelectorPluginConfig"]
-	if !isLegacyImagePolicy && !isLegacyPodNodeSelector {
-		return nil, err
+	for key := range unstructuredData {
```

Comment:
This is not equivalent, and is more strict. This says if any other key exists in the file, disallow it being treated as a legacy config file, which is probably too strict

Reply:
Yes, this is the intention. It will fail if we use a config-file contains any other invalid plugins configuration including non-exist, invalid configuration keys, etc.

Reply:
And that is not what we want. A legacy config file (including one that contained other unknown keys) must continue to load.

Reply:
ok